src/main/frontend/components/query/result.cljs

```diff
@@ -66,7 +67,9 @@
             ;; exclude the current one, otherwise it'll loop forever
             remove-blocks (if current-block-uuid [current-block-uuid] nil)
             transformed-query-result (when query-result
-                                       (db/custom-query-result-transform query-result remove-blocks q))
+                                       (cond-> (db/custom-query-result-transform query-result remove-blocks q)
+                                         (get q :remove-block-children? true)
+                                         tree/filter-top-level-blocks))
```

Comment:
tree filtering was moved here because modifying results outside of `get-query-result` results in the query counts being wrong